### PR TITLE
RavenDB-8184

### PR DESCRIPTION
**Pull request by fitzchak**
RQL auto complete: List nested fields for collections fields
Dot char should not be as part of the identifier but a separator between nested fields.
Include @ char in the identifier for collection: @all_docs.
Show the field type for fields of collections.

### CODE REVIEW

File: src/Raven.Server/Web/Studio/StudioCollectionFieldsHandler.cs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+{
+    public class StudioCollectionFieldsHandler : DatabaseRequestHandler
+    {
+        [ThreadStatic]
+        private static BlittableJsonReaderObject.PropertiesInsertionBuffer _buffers;
+
+        private const int MaxArrayItemsToFetch = 16;
+        
+        [RavenAction("/databases/*/studio/collections/fields", "GET", AuthorizationStatus.ValidUser)]
+        public Task GetCollectionFeilds()
```
Comment (ml054) [inline src/Raven.Server/Web/Studio/StudioCollectionFieldsHandler.cs:22]: typo

File: src/Raven.Studio/typescript/commands/database/documents/getCollectionFieldsCommand.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import database = require("models/resources/database");
+import endpoints = require("endpoints");
+
+class getCollectionFieldsCommand extends commandBase {
+
+    constructor(private database: database, private collectionName: string, private prefix: string) {
+        super();
+    }
+
+    execute(): JQueryPromise<object> {
```
Comment (ml054) [inline src/Raven.Studio/typescript/commands/database/documents/getCollectionFieldsCommand.ts:11]: you can use dictionary<string> type here and in all usages.

File: src/Raven.Server/Web/Studio/StudioCollectionFieldsHandler.cs (hunk 1 above)
Comment (fitzchak) [reply src/Raven.Server/Web/Studio/StudioCollectionFieldsHandler.cs:22]: Fixed

File: src/Raven.Studio/typescript/commands/database/documents/getCollectionFieldsCommand.ts (hunk 2 above)
Comment (fitzchak) [reply src/Raven.Studio/typescript/commands/database/documents/getCollectionFieldsCommand.ts:11]: Changed to `dictionary<string>`. Thanks.

**Pull request merged**